File: SW Project/Customised online judger/problem_set/2017/smart.java

import java.util.Scanner;


public class smart
{
   public boolean isPrime(int num)
   {
      int   k;

      for (k = 2; k <= num / 2; k++)
      {
         if (num % k == 0)
            return false;
      }

      return true;
   }


   public smart()
   {
      Scanner   scanner;
      int       numSets;
      int       k;
      int       num;
      int       i;
      int       j;

      scanner = new Scanner(System.in);

      numSets = scanner.nextInt();
      for (k = 0; k < numSets; k++)
      {
         num = scanner.nextInt();
         System.out.printf("%d ", num);
         if (isPrime(num))
            System.out.printf("0%n");
         else
         {
            for (i = num - 1; !isPrime(i); i--);

            for (j = num + 1; !isPrime(j); j++);

            if (num - i <= j - num)
               System.out.printf("%d%n", num - i);
            else
               System.out.printf("%d%n", j - num);
         }
      }
   }


   public static void main(String[] args)
   {
      new smart();
   }
}

File: SW Project/Customised online judger/problem_set/2017/smart.py

def isPrime(num):
   for k in range(2, num // 2):
      if (num % k == 0):
         return False

   return True


numSets = int(input())

for _ in range(numSets):
   num = int(input())
   print("%d " % num, end='')

   if (isPrime(num)):
      print("0")
   else:
      i = num - 1
      while True:
         if (isPrime(i)):
            break
         else:
            i = i - 1

      j = num + 1
      while True:
         if (isPrime(j)):
            break
         else:
            j = j + 1

      if num - i <= j - num:
         print("%d" % (num - i))
      else:
         print("%d" % (j - num))

